src/backend/distributed/operations/shard_cleaner.c

```diff
@@ -114,7 +171,8 @@ TryDropOrphanedShards(bool waitForLocks)
 	MemoryContext savedContext = CurrentMemoryContext;
 	PG_TRY();
 	{
-		droppedShardCount = DropOrphanedShards(waitForLocks);
+		droppedShardCount = DropOrphanedShardsForMove(waitForLocks);
+		droppedShardCount += DropOrphanedShardsForCleanup();
 	}
 	PG_CATCH();
 	{
```

Comment:
Not this PR's fault, but the PG_CATCH below seems questionable. Errors can happen in any corner, especially out of memory errors. Continuing without going through the proper clean-up logic is dangerous and can often lead to a crash / undefined behaviour. One strategy to make it a bit safer is to use a subtransaction. 

I'm wondering whether we can try to avoid errors in DropOrphanedShardsForCleanup and not add more to this questionable PG_TRY/PG_CATCH, maybe also remove it for DropOrphanedShardsForMove at some point.

Reply:
Fixed. Resolving.